src/Microsoft.Azure.WebJobs.Host/Storage/StorageAccountType.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+namespace Microsoft.Azure.WebJobs.Host.Storage
+{
+    /// <summary>Storage account types.</summary>
+    [Flags]
+    internal enum StorageAccountTypes
```

Comment:
Arose from the storage account attribute stuff.  I'd still say it's useful because you need to pass a list of enums around to do the below matrix, unless if we have extension methods for 'onlyAllow' and 'doNotAllow' (which would not scale if they added another account type with weird restrictions that didnt' fall in these buckets).

- Storage/Dashboard accounts / table bindings / queue bindings == GeneralPurpose
- Blob Triggers / SingletonManager == General Purpose || BlobOnly
- Blob input / output == all types